metrics/web/docs/features/deep_links/01_deep_linking.md

```diff
@@ -0,0 +1,141 @@
+# Deep Linking
+> Feature description / User story.
+
+As a user, I want to pass direct links to project and/or project groups, so that other users can navigate easier.
+
+## Contents
+
+- [**Analysis**](#analysis)
+    - [Feasibility study](#feasibility-study)
+    - [Requirements](#requirements)
+    - [Prerequisites](#prerequisites)
+    - [Landscape](#landscape)
+    - [Prototyping](#prototyping)
+    - [System modeling](#system-modeling)
+
+# Analysis
+> Describe the general analysis approach.
+
+The `Deep Linking` feature improves the navigation experience by allowing users to navigate to a specific resource of an application and to save the application state in a URL.
+
+The analysis stage aims the following goals:
+1. Define the detailed requirements for the deep linking feature;
+2. Investigate existing approaches for the deep linking in the Flutter Web applications.
+3. Define which approach satisfies our requirements: existing external solutions, or the custom-developed approach.
+
+### Feasibility study
+> A preliminary study of the feasibility of implementing this feature.
+
+The `Deep Linking` feature improves the overall user experience in terms of navigation, so this feature makes sense for the project.
+
+To implement the `Deep Linking` feature, we should be able to perform the following actions:
+- Parsing the `query parameters` from the URL;
+- Updating the URL and the `query parameters` in response to any events.
+
+Since `Navigator 2.0` provides the navigation system that allows performing the actions listed above, and the Metrics Web application uses `Navigator 2.0`, it is possible to implement the `Deep Linking` feature in the Metrics Web application.
+
+### Requirements
+> Define requirements and make sure that they are complete.
+
+The deep links implementation approach that we are to select must satisfy the following requirements:
+- Possibility to handle deep links and restore the application's state using them;
+- Possibility to reflect the application's state in deep links as a response to UI events (e.g. filtering, dropdown selections, searching, etc.);
```

Comment:
Would it be great to mention here that we should be able to update the URL query parameters/deep links without changing the current browser history?